pkg/plugins/runtime/gateway/match/hostnames.go

```diff
@@ -35,6 +35,9 @@ func (h *hostname) contains(n hostname) bool {
 }
 
 func makeHostname(name string) hostname {
+	if name == "" {
+		name = "*"
+	}
```

Comment:
we have a bunch of these scattered around the codebase, should we make them point to this function?

Reply:
tried to reduce the `"" => "*"` logic in the codebase